### PR TITLE
Network neighbourhood bug

### PR DESCRIPTION
**Pull request by duchaoyu**
no need to remove the key in the last iteration of the while loop

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/datastructures/network/core/graph.py [hunk 1]
```diff
@@ -1275,8 +1275,6 @@ def neighborhood(self, key, ring=1):
                 temp += self.neighbors(key)
             nbrs.update(temp)
             i += 1
-        if key in nbrs:
-            nbrs.remove(key)
```
Comment (tomvanmele) [inline src/compas/datastructures/network/core/graph.py]: this line does not remove potential duplicates, but rather the original node which should not appear in the list of its own neighbours...

however, there is another problem related to that that is indeed an actual bug (will let you figure it out :)

Comment (duchaoyu) [reply src/compas/datastructures/network/core/graph.py]: oh... yes... It removes the key in the last iteration of the while loop... the last neighbour key in the (i-1)ring.

Comment (tomvanmele) [reply src/compas/datastructures/network/core/graph.py]: the change should actually be
```python
def neighborhood(self, vertex, ring=1):
    nbrs = set(self.neighbors(vertex))
    ...
    if vertex in nbrs:
        nbrs.remove(vertex)
```

Comment (tomvanmele) [reply src/compas/datastructures/network/core/graph.py]: @duchaoyu ping

Comment (tomvanmele) [reply src/compas/datastructures/network/core/graph.py]: please fix so i can merge...

**Review by tomvanmele — APPROVED** [review]
only an entry in the changelog is still missing...

**Comment by duchaoyu** [comment]
hope now it's fine...

**Pull request merged**